tests/specifications/e2e/pipeline_expanded.yaml

```diff
@@ -13,8 +13,13 @@ step_3:
   iterate:
     - implementation:
         name: step_3_python_pandas
-    - implementation:
-        name: step_3_python_pandas
+    - substeps:
```

Comment:
Just so I'm clear, so step_1 is a single step that consists of two sub-steps while step_3 is looped twice where the first loop is a "full" implementation and the second loop is made up of two sub-steps?

Reply:
That is correct!